### PR TITLE
CHAIN-417: arch/bitcoin network docker config and initial rpc clients

### PR DESCRIPTION
**Pull request by bflood366**

### CODE REVIEW

File: docker-compose-arch.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+
```
Comment (bflood366) [inline docker-compose-arch.yaml:1]: for now I made this a different compose file just for starting the arch network/bitcoin, but can merge into one that starts all the other containers

**Review by bholzman — APPROVED** [review]
Looks good!

File: backend/src/main/kotlin/co/chainring/core/blockchain/bitcoin/ArchNetworkClient.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                "get_processed_transaction",
+                RpcParams(txId),
+            ),
+        )
+    }
+
+    fun assignAuthority(request: ArchNetworkRpc.AssignAuthorityParams): ArchNetworkRpc.ProcessedTransaction {
+        return getValue(
+            RpcRequest(
+                "get_processed_transaction",
```
[the remark this answers is not in the document]
Comment (bflood366) [reply backend/src/main/kotlin/co/chainring/core/blockchain/bitcoin/ArchNetworkClient.kt]: fixed

File: backend/src/main/kotlin/co/chainring/core/model/rpc/RpcParams.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                        is Long -> encodeSerializableElement(Long.serializer().descriptor, index, Long.serializer(), it)
+                        is Boolean -> encodeSerializableElement(Boolean.serializer().descriptor, index, Boolean.serializer(), it)
+                    }
+                }
+            }
+            is String -> encoder.encodeString(param)
+            is Int -> encoder.encodeInt(param)
+            is Long -> encoder.encodeLong(param)
+            is Boolean -> encoder.encodeBoolean(param)
+            is ArchNetworkRpc.DeployProgramParams -> ArchNetworkRpc.DeployProgramParams::class.serializer().serialize(encoder, param)
```
[the remark this answers is not in the document]
Comment (bflood366) [reply backend/src/main/kotlin/co/chainring/core/model/rpc/RpcParams.kt]: changed

**Pull request merged**